### PR TITLE
Create new properties section in ci.yaml for Ubuntu bots

### PR DESCRIPTION
**Pull request by yusuf-goog**
Add a new properties section in ci.yaml to which sets os:Ubuntu. Its a copy paste of the existing linux section, with os:Ubuntu.

Issue: #102406

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -11,6 +11,23 @@ enabled_branches:
   - flutter-\d+\.\d+-candidate\.\d+
 
 platform_properties:
+  ubuntu:
```
Comment (yusuf-goog) [inline .ci.yaml:14]: Is this the most elegant way of doing the switchover, or is there another way to perform a staggered switch of builds over to the new Ubuntu hosts.

Comment (CaseyHillers) [reply .ci.yaml:14]: I think the most elegant way would be switching the os over in Linux so you don't need to duplicate builders for the migration.

If there's specific targets failing, or we don't have capacity to do it all at once, we could set os in specific targets.

Comment (yusuf-goog) [reply .ci.yaml:14]: I would like to do the switch over in a staggered manner. So if we can override the os: property in each builder, maybe thats the right way to do it and then switch over the linux: property group and take out the overrides?

Comment (yusuf-goog) [reply .ci.yaml:14]: @godofredoc what do you think?

Comment (keyonghan) [reply .ci.yaml:14]: Could we enable all linux vm tests in staging to run against the ubuntu bots to validate first, see for a couple of days? Then we are pretty confident to switch over to `os: ubuntu`, and inject `os: linux` for targets that are failing.

**Pull request closed without merging**